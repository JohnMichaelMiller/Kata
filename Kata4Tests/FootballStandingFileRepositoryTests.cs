using Microsoft.VisualStudio.TestTools.UnitTesting;
using Kata4;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using ApprovalTests;
using ApprovalTests.Reporters;

namespace Kata4.Tests
{
    [TestClass()]
    [UseReporter(typeof(DiffReporter))]
    public class FootballStandingFileRepositoryTests
    {
        [TestMethod()]
        public void FootballStandingFileRepositoryTest()
        {
            var repo = new FootballStandingFileRepository();
            Approvals.VerifyAll(repo, "Football Standing");
        }
    [TestMethod()]
    public void FootballStandingFileRepositoryHasCorrectMinimumTemperatureDifferential()
    {
        var repo = new FootballStandingFileRepository();
        var result =
            from w in repo
            where -w.GoalDifferential ==
                repo.Min(d => -d.GoalDifferential)
            select w;

        Approvals.VerifyAll(result, "Football Standing");
    }

    }
}