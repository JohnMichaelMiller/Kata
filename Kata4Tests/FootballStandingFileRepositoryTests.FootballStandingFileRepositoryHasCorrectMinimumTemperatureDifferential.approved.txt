Football Standing[0] = Rank:1 Team: Arsenal, Goals For 79, Goals Against: 36, Goal Diff: 43
